File: VB/DataNavigatorInBar/DataNavigatorEdit/NavigatorEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraEditors
Imports System.ComponentModel

Namespace DataNavigatorInBar
	Friend Class NavigatorEdit
		Inherits BaseEdit
		Shared Sub New()
			NavigatorEditRepositoryItem.Register()
		End Sub
		Public Sub New()
			MyBase.New()
			Controls.Add(Properties.EditorNavigator)
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return NavigatorEditRepositoryItem.EditorName
			End Get
		End Property
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As NavigatorEditRepositoryItem
			Get
				Return TryCast(MyBase.Properties, NavigatorEditRepositoryItem)
			End Get
		End Property
		Public Overrides Property EditValue() As Object
			Get
				Return MyBase.EditValue
			End Get
			Set(ByVal value As Object)
				MyBase.EditValue = value
				Properties.EditorNavigator.DataSource = value
			End Set
		End Property
	End Class

End Namespace

File: CS/DataNavigatorInBar/DataNavigatorEdit/NavigatorEdit.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraEditors;
using System.ComponentModel;

namespace DataNavigatorInBar
{
    class NavigatorEdit : BaseEdit
    {
        static NavigatorEdit()
        {
            NavigatorEditRepositoryItem.Register();
        }
        public NavigatorEdit()
            : base()
        {
            Controls.Add(Properties.EditorNavigator);
        }
        public override string EditorTypeName
        {
            get { return NavigatorEditRepositoryItem.EditorName; }
        }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new NavigatorEditRepositoryItem Properties
        {
            get { return base.Properties as NavigatorEditRepositoryItem; }
        }
        public override object EditValue
        {
            get
            {
                return base.EditValue;
            }
            set
            {
                base.EditValue = value;
                Properties.EditorNavigator.DataSource = value;
            }
        }
    }

}
